### DISCUSSION TITLE
Pipeline diagram not working

### DISCUSSION THREAD

**Discussion by efthym** [mortbopet/Ripes#217] [Q&A]
Hi,
I have trouble using the pipeline diagram feature. It just shows the list of instructions. The pipeline stages, etc, don't show up. 

The new window which pops up behaves as a dialogue window (not sure it the term is right): it seems to block the UI of the main window, until it is closed. When I move the mouse anywhere in the main Ripes window to try to click a button (e.g. clock), gnome focuses immediately on the "pipeline diagram" window, so the click is ignored. I also tried the key short cuts, e.g. F5,, but they are also ignored. 

Do you have any thoughts, on how to fix this?  
I'm using the v2.2.4 appimage release and run it on ubuntu 22.04, Gnome. 
Thanks,
Aris

**Comment by mortbopet** [comment]
Sounds like https://github.com/mortbopet/Ripes/issues/209. Should be fixed if you download the continuous release version.

**Reply by efthym** [reply]
Thanks for the quick reply. 
 tried that just after writing the comment, but that doesn't even start on my machine:
```
terminate called after throwing an instance of 'std::out_of_range'
  what():  map::at
```

**Reply by mortbopet** [reply]
That sounds like a proper issue! Mind opening an issue about this?

**Reply by efthym** [reply]
OK. I'm downloading Qt to give a try compiling it from source too.

**Reply by mortbopet** [reply]
Even better! That should hopefully get us on the right track to figuring out the root cause. I know that there is a lot of `.at(...)` usage spread around the codebase which really should have been a `.find(...)` + a bounds check.
